### PR TITLE
Initial GitHub Action to build all artifacts

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This PR adds a new GitHub Action for building all artifacts. This PR currently performs the release builds and keeps all artifacts locally. It is a starting point for further work.

This action requires a user manually start it. It is not triggered automatically.
 
### Issues Resolved

This is only part of #977.
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dapowers87 — APPROVED** [review]

**Review by dinujoh — COMMENTED** [review]
What is the command for user to manually trigger it ? Should we add this to our documentation ?

**Comment by dlvenable** [comment]
@dinujoh , To clarify on the documentation: This release process isn't yet ready. This is just the first commit which shows how the artifacts and Docker image are built locally. I plan to add additional steps to this action to upload these to the staging repositories. This commit sets the baseline job.

If you are curious, you can see the jobs on my fork: https://github.com/dlvenable/data-prepper/actions/workflows/release.yml

(The failed builds are from some experiments which I'm postponing.)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    steps:
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: 14
+    - name: Checkout Data-Prepper
+      uses: actions/checkout@v2
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
```
Comment (cmanning09) [inline .github/workflows/release.yml:18]: Is this step needed? [gradlew](https://github.com/opensearch-project/data-prepper/blob/main/gradlew) already has execute permissions.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - name: Build
+      run: ./gradlew build
+    - name: Archives
+      run: ./gradlew :release:archives:linux:linuxTar -Prelease
+    - name: Docker
```
Comment (cmanning09) [inline .github/workflows/release.yml:23]: Take it or leave it: I would recommend making the names more explicit as `Build`, `Archives`, and `Docker` are a little ambiguous. I like the names you provided in on of the diagrams in the RFC:

- Build jar files
- Build archives || Build tar.gz files
- Build docker image

**Review by dinujoh — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (dlvenable) [reply .github/workflows/release.yml:18]: I asked this same question on another PR recently. It appears all of our GitHub Actions do this. I'll try to experiment to see if this really matters or not.

File: .github/workflows/release.yml (hunk 2 above)
Comment (dlvenable) [reply .github/workflows/release.yml:23]: Thanks! I'll update this in the next PR.

File: .github/workflows/release.yml (hunk 1 above)
Comment (dlvenable) [reply .github/workflows/release.yml:18]: I created a new PR that removes this step for all builds:

https://github.com/opensearch-project/data-prepper/pull/1028

**Pull request merged**